utxo/src/undo.rs

```diff
@@ -0,0 +1,41 @@
+use crate::Utxo;
+use parity_scale_codec::{Decode, Encode};
+
+#[derive(Debug, Eq, PartialEq, Encode, Decode)]
+pub struct TxUndo(Vec<Utxo>);
+
+impl From<Vec<Utxo>> for TxUndo {
+    fn from(list: Vec<Utxo>) -> Self {
+        TxUndo(list)
+    }
+}
+
+impl TxUndo {
+    fn inner(&self) -> &[Utxo] {
+        &self.0
+    }
+
+    fn into_inner(self) -> Vec<Utxo> {
+        self.0
+    }
+}
+
+
+#[derive(Debug, Eq, PartialEq, Encode, Decode)]
+pub struct BlockUndo(Vec<TxUndo>);
+
+impl From<Vec<TxUndo>> for BlockUndo {
```

Comment:
I'm not a big fan of this kind of meaningless conversion... `From` is meant to be kind of isomorphic or epimorphic conversion of data from one type to another (I hope I'm using these names right)... the point is, just because by sheer coincidence the current content of a type is another type, doesn't mean it represents it. To illustrate the issue, if we decide to have the block height baked into Undos in order to test blocks that are not in the main chain, this function will just break.

Reply:
changed to `fn new()`